engine/src/flutter/lib/web_ui/test/engine/platform_view_registry_test.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:test/bootstrap/browser.dart';
+import 'package:test/test.dart';
+import 'package:ui/ui_web/src/ui_web.dart';
+
+void main() {
+  internalBootstrapBrowserTest(() => testMain);
+}
+
+void testMain() {
+  test('platformViewRegistry can be overridden', () {
+    final PlatformViewRegistry overrideRegistry = MockPlatformViewRegistry();
+
+    debugOverridePlatformViewRegistry(overrideRegistry);
+    expect(platformViewRegistry, overrideRegistry);
+
+    debugOverridePlatformViewRegistry(null);
+    expect(platformViewRegistry, isNot(overrideRegistry));
```

Comment:
If you apply the suggestion above, then this test can be updated:
```suggestion
    final PlatformViewRegistry overrideRegistry = MockPlatformViewRegistry();

    final realRegistry = platformViewRegistry;

    debugOverridePlatformViewRegistry(overrideRegistry);
    expect(platformViewRegistry, overrideRegistry);

    debugOverridePlatformViewRegistry(null);
    expect(platformViewRegistry, realRegistry);
```